File: php_src/bourdon/kernel/app/model/importance.php
<?php
require_once(LIB."Model.php");
class importance extends Model
{
    protected $id;
    protected $content;

    public function __construct()
    {
        parent::__construct("importance", "id");
        $this->id = null;
        $this->content = null;
    }

    /**
     * @return null
     */
    public function getId()
    {
        return $this->id;
    }

    /**
     * @param null $id
     */
    public function setId($id)
    {
        $this->id = $id;
    }


    /**
     * @return mixed
     */
    public function getContent()
    {
        return $this->content;
    }

    /**
     * @param mixed $content
     *
     * @return self
     */
    public function setContent($content)
    {
        $this->content = $content;

        return $this;
    }
}

File: src/app/src/main/java/com/example/issac/myapplication/MODEL/Importance.java
package com.example.issac.myapplication.MODEL;


import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;

public class Importance extends Model
{
    protected int id;
    private String content;

    public Importance() {
        super("importance");
        this.content = null;
    }

    public void setContent(String content) {
        this.content = content;
    }

    @Override
    /*
    07/05
    Matthieu (basé sur Boutte)
    Permet d'assigner des attributs d'un objet avec un élément JSON
     */ public void putInObj(JSONObject json) {
        try {
            this.id = json.getInt("id");
            this.content = json.getString("content");                 // ici on assigne la totalité des attributs avec le résultat en JSON (ici modif pour debug req Creation)
        } catch (JSONException e) {
            e.printStackTrace();
        }

    }


    public ArrayList<Importance> JSONArrayToImportances(ArrayList<JSONObject> table) {
        ArrayList<Importance> lesImportances = new ArrayList<Importance>();
        int i;
        for(i=0;i<table.size();i++) {
            Importance lImportance = new Importance();
            lImportance.putInObj(table.get(i));
            lesImportances.add(lImportance);
        }
        return lesImportances;
    }

    public String getContent() {
        return content;
    }


    public int getId() {
        return id;
    }


    public void setId(int id) {
        this.id = id;
    }
}
